protocol/synthetix/contracts/storage/RewardDistribution.sol

```diff
@@ -216,17 +242,29 @@ library RewardDistribution {
 
             // If the current time is beyond the duration, then consider all scheduled value to be distributed.
             // Else, the amount distributed is proportional to the elapsed time.
-            int256 curUpdateDistributedD18 = self.scheduledValueD18;
+            int256 curUpdateDistributedD18;
             if (curTime < self.start + self.duration) {
                 // Note: Not using an intermediate time ratio variable
                 // in the following calculation to maintain precision.
                 curUpdateDistributedD18 =
                     (curUpdateDistributedD18 * (curTime - self.start).toInt()) /
                     self.duration.toInt();
-            }
+						} else {
+							// if we have reached the end of the distribution, we should cycle through to the next distribution (if any)
+							curUpdateDistributedD18 = self.scheduledValueD18;
+							self.start = self.nextStart;
+							self.duration = self.nextDuration;
+							self.scheduledValueD18 = self.nextScheduledValueD18;
+							self.nextStart = 0;
+							self.nextDuration = 0;
+							self.nextScheduledValueD18 = 0;
+
+							// start processing the new entry also
+							valuePerShareChangeD18 = updateEntry(self, totalSharesAmountD18);
```

Comment:
Recursive call? 
Can we extract functionality into internal `_updateEntry` and reuse it depending on current conditions?

Reply:
the call itself already acts as an internal role that also happens to be used externally whenever `totalShares` of a distribution changes.

there is no benefit that I can see because:
* all the functionality of the existing `updateEntry` function is being used by this recursive call
* this funtion in and of itself is used both internally (through this recursive call, and call through `distribute` function in this file), as well as externally (when the distributions `totalShares` changes, it is supposed to be called by users of the library)